src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java

```diff
@@ -380,29 +388,89 @@ private Conditions idQueryFromTuple(TarantoolTuple tuple, TarantoolPersistentEnt
+
+        List<Object> indexPartValues = getIndexPartValues(source, entity);
+        return createIndexEqualsConditionFromParts(indexPartValues);
+    }
+
+    private <T> List<Object> getIndexPartValues(T source, TarantoolPersistentEntity<?> entity) {
+        return entity.hasTarantoolIdClassAnnotation() ?
+                getIndexPartsFromCompositeIdValue(source, entity)
+                : Collections.singletonList((Object) source);
+    }
+
+    private List<Object> getIndexPartsFromCompositeIdValue(Object idValue, TarantoolPersistentEntity<?> entity) {
+        //for each property get field name and map to idValue bean property value
+        List<Object> list = new ArrayList<>();
+        Class<?> idClass = entity.getTarantoolIdClass();
+        Assert.notNull(idClass, "TarantoolIdClass is null for entity " + entity);
+
+        for (Field f : idValue.getClass().getDeclaredFields()) {
+            TarantoolPersistentProperty property = entity.getPersistentProperty(f.getName());
+            Assert.notNull(property, "Property " + f.getName() + "does not exist in entity " + entity);
+            if(!property.isIdProperty()) {
+                throw new TarantoolTypeConversionException(idValue.getClass(), idClass);
             }
 
-            PersistentPropertyAccessor<?> propertyAccessor = entity.getPropertyAccessor(source);
-            idValue = propertyAccessor.getProperty(idProperty);
-            if (idValue == null) {
-                throw new MappingException("ID property value is null");
+            ReflectionUtils.makeAccessible(f);
+            try {
+                list.add(f.get(idValue));
+            } catch (IllegalAccessException e) {
+                throw new TarantoolEntityMappingException(entity.getClass(), e);
             }
         }
-        Optional<Class<?>> basicTargetType = converter.getCustomConversions().getCustomWriteTarget(idValue.getClass());
-        if (basicTargetType.isPresent()) {
-            idValue = converter.getConversionService().convert(idValue, basicTargetType.get());
-        }
+        return list;
+    }
+
+    private Conditions createIndexEqualsConditionFromParts(List<Object> indexPartValues) {
+        List<Object> indexPartValuesConverted = indexPartValues.stream()
+                .map( obj -> {
+                    Optional<Class<?>> basicTargetType = converter
+                            .getCustomConversions()
+                            .getCustomWriteTarget(obj.getClass());
+
+                    return basicTargetType.isPresent() ?
+                            converter.getConversionService().convert(obj, basicTargetType.get())
+                            : obj;
+                })
+                .collect(Collectors.toList());
 
-        return Conditions.indexEquals(0, Collections.singletonList(idValue));
+        final int PK_INDEX_ID = 0;
+        return Conditions.indexEquals(PK_INDEX_ID, indexPartValuesConverted);
```

Comment:
Add an issue to cartridge-driver about adding such constant to Conditions. BTW, there is https://tarantool.github.io/cartridge-java/io/tarantool/driver/protocol/TarantoolIndexQuery.html#PRIMARY already.

Reply:
Created https://github.com/tarantool/cartridge-java/issues/92